shopfloor/services/location_content_transfer.py

```diff
@@ -0,0 +1,922 @@
+
+        If the move has other move lines / package levels it has to be split
+        so we can post only this part.
+
+        After the destination is set, the move is set to done.
+
+        Beware, when _action_done() is called on the move, the normal behavior
+        of Odoo would be to create a backorder transfer. We don't want this or
+        we would have a backorder per move. The context key
+        ``_sf_no_backorder`` disables the creation of backorders, it must be set
+        on all moves, but the last one of a transfer (so in case something was not
+        available, a backorder is created).
+
+        Transitions:
+        * scan_destination: invalid destination or could not
+        * start_single: continue with the next package level / line
+        * start: if there is no more package level / line to process
+        """
+        location = self.env["stock.location"].browse(location_id)
+        package_level = self.env["stock.package_level"].browse(package_level_id)
+        if not location.exists() or not package_level.exists():
+            return self._response_for_start(message=self.msg_store.record_not_found())
+        search = self.actions_for("search")
+        scanned_location = search.location_from_scan(barcode)
+        if not scanned_location:
+            return self._response_for_scan_destination(
+                location, package_level, message=self.msg_store.no_location_found()
+            )
+        if not scanned_location.is_sublocation_of(
+            package_level.picking_id.picking_type_id.default_location_dest_id
+        ):
+            return self._response_for_scan_destination(
+                location,
+                package_level,
+                message=self.msg_store.dest_location_not_allowed(),
+            )
+        if not scanned_location.is_sublocation_of(package_level.location_dest_id):
+            if not confirmation:
+                return self._response_for_confirm_scan_destination(
+                    location, package_level
+                )
+        package_move_lines = package_level.move_line_ids
+        package_moves = package_move_lines.mapped("move_id")
+        for package_move in package_moves:
+            # Check if there is no other lines linked to the move others than
+            # the lines related to the package itself. In such case we have to
+            # split the move to process only the lines related to the package.
+            other_move_lines = package_move.move_line_ids - package_move_lines
+            if other_move_lines:
+                qty_to_split = sum(other_move_lines.mapped("product_uom_qty"))
+                backorder_move_id = package_move._split(qty_to_split)
+                backorder_move = self.env["stock.move"].browse(backorder_move_id)
+                backorder_move.move_line_ids = other_move_lines
+                backorder_move._action_assign()
+        if package_move_lines == package_moves.mapped("move_line_ids"):
```

Comment:
It was for the case where one move uses 1 package and 1 another move.line to be satisfied, so that the move lines related to the package has to be the same than the move lines of the moves related to the package.
Obviously the code above should ensure that we have 1 move for the current package, so yes it's should be always true... Yup better to remove it.